core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -219,16 +219,14 @@ class AdminManager(val config: KafkaConfig,
           case (topicPartition, replicas) => topicPartition.partition -> replicas
         }
         if (existingAssignment.isEmpty)
-          throw new UnknownTopicOrPartitionException(s"The topic '$topic' does not exist")
+          throw new UnknownTopicOrPartitionException(s"The topic '$topic' does not exist.")
 
         val oldNumPartitions = existingAssignment.size
         val newNumPartitions = newPartition.totalCount
         val numPartitionsIncrement = newNumPartitions - oldNumPartitions
         if (numPartitionsIncrement < 0) {
           throw new InvalidPartitionsException(
             s"Topic currently has $oldNumPartitions partitions, which is higher than the requested $newNumPartitions.")
-        } else if (numPartitionsIncrement == 0) {
-          throw new InvalidPartitionsException(s"Topic already has $oldNumPartitions partitions.")
         }
```

Comment:
Do we want to restore these 2 lines too?

Reply:
Yes, we do. The error message is more succinct too. Now fixed.